app/controllers/manage_cmr/permissions_controller.rb

```diff
@@ -148,12 +146,28 @@ def destroy
 
   private
 
+  def collection_permission_params
+    # because there are nested params, when they go through strong parameters
+    # the nested params become a ActionController Parameters object
+    # so when grabbing the nested params we need to call to_h on them or some
+    # hash methods will cause an error
```

Comment:
do the `to_h` here instead of needing to call it in all the other places.

Reply:
@ryanmiller-1 , I could not get this to work. Even if I returned a param with the nested params turned into a hash out of this method, somehow the nested hashes we expected to grab out of it still turned into a parameters object. I can show you if you are interested, but I don't think we can make the fix.

Reply:
Rails really doesn't want you to do this.  They did provide a 'to_unsafe_h' which still doesn't do exactly what you want.      We have a ticket to avoid these problems by having a method that creates a hash from params using a passed in structure of the keys that you want to permit.  It is probably time to fix that ticket so all the controllers work the same way.

Reply:
The suggestion was to get the permitted parameters as a hash because the parameter class that they are does not implement all the hash methods (I assume Charles can provide a specific example as he found it necessary to .to_h in some places). I don't think that is counter to the spirit of the strong params...

Reply:
@jdf-eos is right, there is some Rails magic happening that won't allow us to make nested params into a straight hash. Tried doing that in the strong params method here, as well as assigning to an ivar in the compute method. In both cases, the nested params after manipulation still came out as a Parameters object. So basically, we can't avoid having to call `to_h` in the places we are grabbing the nested params.